www/js/typechecker.js

```diff
@@ -0,0 +1,107 @@
+/* global define */
+/**
+ *Dragoon Project
+ *Arizona State University
+ *(c) 2014, Arizona Board of Regents for and on behalf of Arizona State University
+ *
+ *This file is a part of Dragoon
+ *Dragoon is free software: you can redistribute it and/or modify
+ *it under the terms of the GNU Lesser General Public License as published by
+ *the Free Software Foundation, either version 3 of the License, or
+ *(at your option) any later version.
+ *
+ *Dragoon is distributed in the hope that it will be useful,
+ *but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+ *GNU Lesser General Public License for more details.
+ *
+ *You should have received a copy of the GNU Lesser General Public License
+ *along with Dragoon.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+define([
+	"dojo/_base/array", "dojo/dom",
+	"dijit/popup", "dijit/TooltipDialog"
+], function(array, dom, popup, TooltipDialog){
+
+	return {
+
+		// In case any tool tips are still open.
+		myTooltipDialog: new TooltipDialog({
+			style: "width: 150px;",
+			content: "Use decimals instead of percent."
+		}),
+
+		// Tool Tip for indicating non numeric data is not accepted
+		myTooltipDialog2: new TooltipDialog({
+			style: "width: 150px;",
+			content: "Non-numeric data not accepted"
+		}),
+
+		dialogs: [this.myTooltipDialog, this.myTooltipDialog2],
+
+		closePops: function(){
+			array.forEach(this.dialogs, function(dialog){
+				popup.close(dialog);
+			});
+		},
+
+		checkInitialValue: function(nodeID, lastInput){
```

Comment:
I think, it would be better if we modularize this into two separate functions, in case something comes up in the future. One function can be check numeric value and the other can check last input value. Giving it appropriate names and it starts becoming a general library of small utilities or validation functions.